File: models/api/prompt.ts
import { BaseApiModel } from 'app/models/api/response';

export enum PromptType {
    fillNewProperties = 'fillNewProperties',
    recurringPaymentFailed = 'recurringPaymentFailed',
    avatarReminder = 'avatarReminder',
    availabilityReminder = 'availabilityReminder',
    noAvailabilityReminder = 'noAvailabilityReminder',
    negativeReview = 'negativeReview',
    positiveReview = 'positiveReview',
    positiveReviewEkomi = 'positiveReviewEkomi',
    positiveReviewTrustpilot = 'positiveReviewTrustpilot',
    positiveReviewGoogle = 'positiveReviewGoogle',
    firstRecommendation = 'firstRecommendation',
    avatarOverlay = 'avatarOverlay',
}
export const allPromptTypes = Object.values(PromptType);

export class Prompt extends BaseApiModel {
    type: PromptType;
    delay = 0;

    static promptWithType(promptType: PromptType, delay = 0) {
        const prompt = new Prompt();
        prompt.type = promptType;
        prompt.delay = delay;
        return prompt;
    }
}

File: ios-app/models/Prompt.swift
import Foundation

enum PromptType: String {
    case avatarReminder
    case availabilityReminder
    case noAvailabilityReminder
    case negativeReview
    case positiveReview
    case firstRecommendation
    case newApplication
    case avatarOverlay
}

class Prompt: JsonApiMappable {
    var type: PromptType
    var delay: Double?

    required init(data: JsonData, includes: [[String: Any]]? = nil) throws {
        let attributes = data.attributes
        if let type = try PromptType(rawValue: attributes.valueForKey("type")) {
            self.type = type
        } else {
            throw ParsingError.general
        }
        delay = try? attributes.valueForKey("delay")
    }
}

extension Notification.Name {
    static let jobPostingStateChanged = Notification.Name("jobPostingStateChanged")
}
